### DISCUSSION TITLE
"No primary field found. Please create and mount one before calling 'createCardToken'. Ignoring call..."

### DISCUSSION THREAD

**Discussion by r9software** [mercadopago/sdk-js#205] [Q&A] [answered]
I'm working with Vue I have loaded successfully the MP elements with core following this which I believe is incomplete
https://www.mercadopago.com.mx/developers/es/docs/checkout-api/integration-configuration/card/integrate-via-core-methods
because the method createSelectOptions is called multiple times but is not in there, I found it in other doc.

Once I call the payment and try to create the card token I receive this error but I haven't found what does it mean.

**Comment by r9software** [comment]
The issue was that it requires the card element to be a div

**Comment by icaldana** [comment, marked as answer]
Hello @r9software! I see the issue is already closed but I'd like to add more info. First of all you are correct, this message is sent when the field wasn’t properly initialised as [mentioned on the documentation](https://www.mercadopago.com.mx/developers/es/docs/checkout-api/integration-configuration/card/integrate-via-core-methods#editor_6).

```
    const cardNumberElement = mp.fields.create('cardNumber', {
      placeholder: "Número de la tarjeta"
    }).mount('form-checkout__cardNumber');
```

Secondly, I recommend you to use [Payment Brick](https://www.mercadopago.com.mx/developers/es/live-demo/payment-brick) instead of Core Methods, since you will have all the logic you mentioned plus the UI for free.
